### PR TITLE
Docs: Recipes / Code-Coverage - AVA+NYC on watch using nodemon

### PR DESCRIPTION
**Pull request by revelt**
Let's add some examples of how to run both AVA and NYC on watch mode. Also, I fixed few English grammar errors in the original copy.

### CODE REVIEW

File: docs/recipes/code-coverage.md [hunk 1]
```diff
@@ -85,11 +85,11 @@ Since it is unlikely you want `inline` source maps in your production code. You
 
 > WARNING: `BABEL_ENV=production` does not work on Windows, you must use the `set` keyword  (`set BABEL_ENV=production`).  For cross platform builds, check out [`cross-env`].
 
-Note that the build script really has very little to do with AVA, and is just a demonstration of how to use Babel's `env` configuration to manipulate your config so it's compatible with AVA.
+Note that the build script has very little to do with AVA and is just a demonstration of how to use Babel's `env` configuration to manipulate your config, so it's compatible with AVA.
 
 ### Use the Babel require hook
 
-To use the Babel require hook, add `babel-core/register` to the `require` section of you AVA config in `package.json`.
+To use the Babel `require` hook, add `babel-core/register` to the `require` section of you AVA config in `package.json`.
```
Comment (novemberborn) [inline docs/recipes/code-coverage.md]: [Babel just uses *require hook*](https://babeljs.io/docs/usage/babel-register/).

While we're updating this though, we should recommend using `babel-register`, not `babel-core/register` which is undocumented (and, without verifying, might very well be going away in Babel 7).

File: docs/recipes/code-coverage.md [hunk 2]
```diff
@@ -170,6 +170,28 @@ after_success:
...
+
+```bash
+$ npm install nodemon --save-dev
+```
+
+Then, create a new npm scripts entry, let's say, `watch`:
+
+```json
+"scripts": {
+	"coverage": "nyc report --reporter=text-lcov | coveralls",
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: This is unrelated to what you're trying to show.

File: docs/recipes/code-coverage.md [hunk 3]
```diff
@@ -170,6 +170,28 @@ after_success:
...
+	"test": "standard && nyc --reporter=html --reporter=text ava",
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: `standard && ` too.

And put the reporter option as package.json config instead CLI flags. (See the `nyc` docs).

File: docs/recipes/code-coverage.md [hunk 4]
```diff
@@ -170,6 +170,28 @@ after_success:
...
+	"watch": "nodemon --quiet --watch ./ --exec npm run test"
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: `./` => `.`

File: docs/recipes/code-coverage.md [hunk 5]
```diff
@@ -170,6 +170,28 @@ after_success:
 
 Your coverage report will then appear on coveralls shortly after Travis completes.
 
+## Integrating with AVA's Watch mode
+
+While AVA's watch mode [won't](https://github.com/avajs/ava/issues/1100) integrate with `nyc` directly out of the box, you can achieve that using [nodemon](https://github.com/remy/nodemon).
+
+First, install `nodemon` as a dev dependency:
+
+```bash
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: `bash` => `console`

File: docs/recipes/code-coverage.md [hunk 6]
```diff
@@ -170,6 +170,28 @@ after_success:
 
 Your coverage report will then appear on coveralls shortly after Travis completes.
 
+## Integrating with AVA's Watch mode
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: Watch => watch

File: docs/recipes/code-coverage.md (hunk 1 above)
Comment (revelt) [inline docs/recipes/code-coverage.md]: Thank you. I extended the tick to include the word `hook` too. Let's do `babel-register` as a separate commit, let's get `nodemon` example sorted first. What do you think?

File: docs/recipes/code-coverage.md (hunk 6 above)
Comment (revelt) [inline docs/recipes/code-coverage.md]: Reworded the title completely. Like Mark noted, it was not accurate actually, we can't say "AVA's watch" because it's "nodemon's watch". Please review.

File: docs/recipes/code-coverage.md (hunk 5 above)
Comment (revelt) [inline docs/recipes/code-coverage.md]: Done, good spot. I remember some of my editors had issues in the past recognising `console` but it's probably an old reflex. Thank you. Please review.

File: docs/recipes/code-coverage.md (hunk 2 above)
Comment (revelt) [inline docs/recipes/code-coverage.md]: I considered your point, slept over it and considered again. This line is called by `.travis.yml`. Without it, Travis won't be able to ping Coveralls. We _can_ omit it, but then we should note aside that `scripts` snippet is incomplete. Some people might think it's complete, paste `gung-ho` and their Coveralls won't update. Being a lazy person, I like complete snippets, ones without missing bits.

I'm happy to remove this line, just say no and I'll remove it.

File: docs/recipes/code-coverage.md (hunk 3 above)
Comment (revelt) [inline docs/recipes/code-coverage.md]: Done. Please review. I tested the separated setup on one of my libraries, seems fine on mine. Thank you, that's a new thing learned. 👍

File: docs/recipes/code-coverage.md (hunk 4 above)
Comment (revelt) [inline docs/recipes/code-coverage.md]: Done good spot. Please review.

File: docs/recipes/code-coverage.md (hunk 2 above)
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: It's not used by `.travis.yml`. Read the above `.travis.yml` code.

File: docs/recipes/code-coverage.md [hunk 7]
```diff
@@ -170,24 +170,30 @@ after_success:
...
 ```json
 "scripts": {
 	"coverage": "nyc report --reporter=text-lcov | coveralls",
-	"test": "standard && nyc --reporter=html --reporter=text ava",
-	"watch": "nodemon --quiet --watch ./ --exec npm run test"
+	"test": "xo && nyc ava",
+	"watch": "nodemon --quiet --watch . --exec npm run test"
 },
+"nyc": {
+	"reporter": [
```
Comment (sindresorhus) [inline docs/recipes/code-coverage.md]: Incorrect indentation here.

**Comment by revelt** [comment]
Hi Sindre, guys, I'll take down the PR, but feel free to amend, reuse or replace the text whichever way you want (or drop it completely).

**Comment by sindresorhus** [comment]
@revelt Why?

**Comment by revelt** [comment]
@sindresorhus that's fine, feel free to fix it and publish. It will be more effective, less backwards forwards. Seriously, no hurt feelings. I'll help all I can and keep contributing.

**Comment by sindresorhus** [comment]
No, I mean, why not just finish it? You only had a couple of super minor changes left. I'm curious, as I often see people just not finishing almost done PRs and I still don't understand why.

**Comment by revelt** [comment]
You know Monopoly, it was a popular game but its rules inadvertently alienated the families. It was people winning against people, not a family winning together against imaginary "Monopoly". I find GitHub a bit of a similar thing. Documentation editing on OS projects should be like a happy Google Docs editing session: somebody writes up something new, others jump in and everybody edit the damn thing TOGETHER, quickly, happily and in live mode. Like a public Google Doc. If admin/maintainer does not like certain character, he just edits it over and happy days. As opposed to formal "requests" and rubbing in to original poster's ego (no offence, I'm talking in general). I know I was not entirely exact on some places, but "requests" to amend make it public which hurts. It's like Monopoly, giving away your albeit imaginary money and failing in front of siblings. Also, you know, any change, even to readme, triggers CI on GitHub. They should review the whole documentation concept a bit - how people work together.

**Comment by sindresorhus** [comment]
> It was people winning against people, not a family winning together against imaginary "Monopoly".

But open-source **is** "a family winning together"! 🤗

> As opposed to formal "requests" and rubbing in to original poster's ego

I find it 😔 that you consider a PR review "rubbing it in". Rather look at it as an opportunity to learn or see other ways to solve problems. Your PR was actually correct, it was just not according to our preferences and required some minor tweaks, which you couldn't have known, and it's as expected. Very few, or close to no PRs are merged without changes.

I could have done the changes myself of course. It would have taken much less time. But I value pull request reviews, both as a submitter and as a reviewer. I often learn things as a reviewer on PRs. I suggest some change and the submitter elaborates on why the existing way is better. I don't always request the best changes, but by requesting something, I start a conversation about it.

I would recommend leaving out "ego" and "emotions" when doing open-source. Yes, I know it's hard.

**Comment by StoneCypher** [comment]
linters often prevent hurt feelings over formatting rules
